### DISCUSSION TITLE
What does `await next()` do?

### DISCUSSION THREAD

**Discussion by navtoj** [honojs/hono#803] [Q&A]
I saw `await next()` being used in the docs but couldn't find any info about when and why it should be used. Does anyone know?

**Comment by yusukebe** [comment]
`await next()` means "dispatch handler inside". For example, if you write:

```ts
app.get('*', async (c, next) => {
  console.log('before')
  await next()
  console.log('after')
})

app.get('/', (c) => {
  console.log('from handler')
  return c.text('Hello')
})
```

then, it will print the following

```
before
from handler
after
```

**Reply by navtoj** [reply]
@yusukebe If there are multiple `app.get('...'` handlers present for different routes, which one gets dispatched inside the `app.get('*'` catchall?
